File: Monsoon Break/gemini2.cpp
#include <iostream>
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

void solve() {
    int N;
    cin >> N;
    string S;
    cin >> S;
    int K;
    cin >> K;

    int max_len = 0;
    int w_count = 0;
    int left = 0;

    for (int right = 0; right < N; ++right) {
        if (S[right] == 'W') {
            w_count++;
        }
        while (w_count > K) {
            if (S[left] == 'W') {
                w_count--;
            }
            left++;
        }
        max_len = max(max_len, right - left + 1);
    }

    cout << max_len << "\n";
}

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    int T;
    if (cin >> T) {
        while (T--) {
            solve();
        }
    }
    return 0;
}

File: Monsoon Break/gemini2.py
import sys

def solve():
    input = sys.stdin.read
    data = input().split()
    
    if not data:
        return
        
    T = int(data[0])
    idx = 1
    
    out = []
    for _ in range(T):
        N = int(data[idx])
        S = data[idx+1]
        K = int(data[idx+2])
        idx += 3
        
        max_len = 0
        w_count = 0
        left = 0
        
        for right in range(N):
            if S[right] == 'W':
                w_count += 1
            
            while w_count > K:
                if S[left] == 'W':
                    w_count -= 1
                left += 1
                
            max_len = max(max_len, right - left + 1)
            
        out.append(str(max_len))
        
    print('\n'.join(out))

if __name__ == '__main__':
    solve()